### PR TITLE
Preselect document if only one doc available

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
In the current flow, you can customise the number of documents you want the users to select. If there is only one type in the list, users still have to select the document which is not an ideal UX. See current behaviour below
Screen Shot 2019-03-27 at 10 03 04

# Solution
If there is only one type of document in the list, user should not see the document selection screen and should be taken straight to the capture screen.
Additionally, I've made the titles on the capture screens more specific, so the user knows what type of document they need to provide.

Additionally, the demo app can be open with a query string `?oneDoc=true` to test this feature.

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [x] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [x] Have tests passed locally?
- [x] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

File: src/index.js [hunk 1]
```diff
@@ -30,11 +30,30 @@ class Container extends Component {
...
+  handledPreselectedDocument = (steps, actions) => {
+    const isDocument = (step) => step.type === 'document'
+    const documentStep = Array.find(steps, isDocument)
+    const hasDocTypes = documentStep && documentStep.options && documentStep.options.documentTypes
+    if (hasDocTypes) {
+      const docTypes = documentStep.options.documentTypes
+      const enabledDocTypes = Object.keys(docTypes).filter((type) => docTypes[type])
+      if (enabledDocTypes.length === 1) {
+        actions.setDocumentType(enabledDocTypes[0])
+        actions.hasPreselectedDoc(true)
```
Comment (rfreitas) [inline src/index.js]: why is this going to store? It's not really state that will be modified internally.

File: src/components/Router/index.js [hunk 2]
```diff
@@ -303,8 +304,8 @@ class HistoryRouter extends Component {
 
   buildComponentsList =
     ({flow},
-    {documentType, steps, options: {mobileFlow}}) =>
-      componentsList({flow, documentType, steps, mobileFlow});
+    {documentType, hasPreselectedDoc, steps, options: {mobileFlow}}) =>
```
Comment (rfreitas) [inline src/components/Router/index.js]: the information of whether there is a preselected document is in the `options` isn't it? Not sure why it needs to be passed as an additional arguments. Feels redundant.

File: src/index.js [hunk 3]
```diff
@@ -30,11 +30,30 @@ class Container extends Component {
...
+      this.handledPreselectedDocument(steps, actions)
+    }
+  }
+
+  handledPreselectedDocument = (steps, actions) => {
+    const isDocument = (step) => step.type === 'document'
+    const documentStep = Array.find(steps, isDocument)
+    const hasDocTypes = documentStep && documentStep.options && documentStep.options.documentTypes
+    if (hasDocTypes) {
+      const docTypes = documentStep.options.documentTypes
```
Comment (rfreitas) [inline src/index.js]: `hasDocTypes` is the same as `docTypes` inside this `if`. Might as well use it, but if so it should be renamed.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Router/StepComponentMap.js [hunk 4]
```diff
@@ -31,6 +31,16 @@ const shouldUseVideo = steps => {
   return (faceOptions || {}).requestedVariant === 'video' && window.MediaRecorder
 }
 
+const hasPreselectedDocument = (steps) => enabledDocuments(steps).length === 1
+
+// This logic should not leave here.
```
Comment (joninvski) [inline src/components/Router/StepComponentMap.js]: `s/leave/live`

File: src/components/Router/StepComponentMap.js [hunk 5]
```diff
@@ -47,7 +47,8 @@ const captureStepsComponents = (documentType, mobileFlow, steps) => {
 
 const createIdentityDocumentComponents = (documentType) => {
   const double_sided_docs = ['driving_licence', 'national_identity_card']
-  const frontDocumentFlow = [SelectIdentityDocument, FrontDocumentCapture, DocumentFrontConfirm]
+  const frontCaptureComponents = [FrontDocumentCapture, DocumentFrontConfirm]
```
Comment (joninvski) [inline src/components/Router/StepComponentMap.js]: :+1:

File: README.md [hunk 6]
```diff
@@ -344,7 +344,7 @@ A number of options are available to allow you to customise the SDK:
   The custom options are:
   - documentTypes (object)
 
-  The list of document types visible to the user can be filtered by using the `documentTypes` option. The default value for each document type is `true`.
+  The list of document types visible to the user can be filtered by using the `documentTypes` option. The default value for each document type is `true`. If only one document type is selected, the user will not see the document selection screen, instead they will see the capture screen straight away.
```
Comment (joninvski) [inline README.md]: Not sure, but if we replace

```
If only one document type is selected, the....
```

to:

```
If only one document type is set, the....
```

Isn't it clearer?

File: test/features/sdk_flow.feature [hunk 7]
```diff
@@ -237,3 +237,9 @@ Feature: SDK File Upload Tests
       Then I should see 3 document_select_buttons ()
       When I click on passport ()
       Then page_title should include translation for "cross_device.intro.document.title"
+
+    Scenario Outline: I should be able to submit a document without seeing the document selector screen
+      Given I navigate to the SDK with one document type
+      When I click on primary_button ()
+      Then I should not see document_select_buttons ()
+      Then page_title should include translation for "capture.passport.front.title"
```
Comment (joninvski) [inline test/features/sdk_flow.feature]: :+1:

File: README.md (hunk 6 above)
Comment (stefaniacardenas) [reply README.md]: I've changed it to ```If `documentTypes` only includes one document type, users will not see the document selection screen, instead they will be taken to the capture screen straight away.```

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-606-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://606-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**